src/cpp/threadpool.cpp

```diff
@@ -0,0 +1,164 @@
+#include <iostream>
+#include <string>
+#include <thread>
+#include <mutex>
+#include <condition_variable>
+#include <chrono>
+
+#include "threadpool.h"
+#include "srexception.h"
+
+using namespace SmartRedis;
+using namespace std::chrono_literals;
+
+// TIME_THREADPOOL
+// Enable this flag to display timings for the threadpool activity.
+// Currently, this will be to screen, but eventually it will go to
+// the SmartRedis log
+#undef TIME_THREADPOOL
```

Comment:
Are we controlling behavior via CPP flags elsewhere? As much as it makes me uncomfortable, perhaps this could also be a different set of environment variables.

Reply:
This is never something that an end user would have any reason to do, just a developer. When we get logging implemented and add a debug flag, this code can be switched to be dependent on the debug flag, but until then, this keeps the code around without interfering with production usage.